File: algorithms/2_medium/S/00540_single_element_in_a_sorted_array/solution.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        const size_t len{nums.size()};

        for (size_t i{0}; i + 1 < len; i += 2) {
            if (nums[i] != nums[i + 1]) {
                return nums[i];
            }
        }

        return nums[len - 1];
    }
};

File: algorithms/2_medium/S/00540_single_element_in_a_sorted_array/solution.rs
impl Solution {
    pub fn single_non_duplicate(nums: Vec<i32>) -> i32 {
        let len = nums.len();

        for i in (0..(len - 1)).step_by(2) {
            if nums[i] != nums[i + 1] {
                return nums[i];
            }
        }

        nums[len - 1]
    }
}
